projects/spectator/src/lib/internals.ts

```diff
@@ -70,7 +70,15 @@ export class Spectator<C> {
   query<R extends Element>(directiveOrSelector: string | DOMSelector): R;
   query<R>(directiveOrSelector: Type<any>, options?: { read }): R;
   query<R>(directiveOrSelector: Type<any> | DOMSelector | string, options: { read } = { read: undefined }): R {
-    return _getChild<R>(this.debugElement)(directiveOrSelector, options);
+    try {
+      return _getChild<R>(this.debugElement)(directiveOrSelector, options);
+    } catch (err) {
+      if (err instanceof SpectatorDebugElementNotFoundError) {
+        return null;
+      } else {
+        throw err;
+      }
+    }
```

Comment:
What about queryAll?

Reply:
Tested that one and it worked fine so no change is required.

Test for it is here: https://github.com/NetanelBasal/spectator/pull/86/files#diff-e6572426b83a2437724cd9a71d19a7deR92